core-common/src/main/java/org/zalando/nakadi/domain/NakadiRecord.java

```diff
@@ -108,4 +109,12 @@ public NakadiRecord setMetadata(final NakadiMetadata metadata) {
         return this;
     }
 
+    public String getPartition() {
+        return partition;
+    }
+
+    public NakadiRecord setPartition(final String partition) {
+        this.partition = partition;
```

Comment:
Can we set partition field of the `metadata` object directly instead?

Reply:
The events of event type with category `undefined` don't have a `metadata` AFAIK.
This means that to set partition field of `metadata` object directly, the class responsible for partitioning has to be aware of the category of the event-type.

Reply:
But it doesn't seem like this method is actually ever called?  Or am I looking in the wrong place?

Reply:
No, this is not yet called. 
This part will follow with its own ticket. 
Basically there is a set of `Check` implementations that are applied to the `NakadiRecorrd`. 
Enriching metadata is the one handled in this ticket, assigning correct partition is a different one.